src/features/vote/SubmitModal.tsx

```diff
@@ -0,0 +1,156 @@
+  HTMLElement,
+  PropsWithChildren<Props>
+> = (
+  {
+    isOpen,
+    close,
+    modalState,
+    showModalSummary,
+    setModalState,
+    reset,
+    handleSubmit,
+    onSubmit,
+  },
+  externalRef
+) => {
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={() => close()} ref={externalRef}>
+        <ModalWrapper>
+          <div className="icon-wrapper">
+            {modalState === "pending" ? (
+              <div className="modal__ico modal__ico-animate">
+                <UnlockedIcon className="unlocked-icon" />
+
+                <div className="modal__ico-container">
+                  <div className="modal__ico-halfclip">
+                    <div className="modal__ico-halfcircle modal__ico-clipped"></div>
+                  </div>
+
+                  <div className="modal__ico-halfcircle modal__ico-fixed"></div>
+                </div>
+              </div>
+            ) : modalState === "success" ? (
+              <LockedIconCommitted className="unlocked-icon" />
+            ) : (
+              <UnlockedIcon className="unlocked-icon" />
+            )}
+          </div>
+
+          {modalState === "pending" ? (
+            <h3 className="header">Committing Votes...</h3>
+          ) : modalState === "success" ? (
+            <h3 className="header">Votes successfully committed</h3>
+          ) : (
+            <h3 className="header">Ready to commit these votes?</h3>
+          )}
+
+          {showModalSummary().length
+            ? showModalSummary().map((el, index) => {
+                return (
+                  <div className="vote-wrapper" key={index}>
+                    <div>{el.identifier}</div>
+                    <div>{el.value}</div>
+                  </div>
+                );
+              })
+            : null}
+          <div
+            className={`button-wrapper ${
+              modalState === "pending" ? "pending" : ""
+            }`}
+          >
+            {modalState === "init" ||
+            modalState === "pending" ||
+            modalState === "error" ? (
+              <>
+                <Button onClick={() => close()} variant="primary">
+                  Not Yet
+                </Button>
+                <Button
+                  type="submit"
+                  onClick={handleSubmit(onSubmit)}
+                  variant="secondary"
+                >
+                  I'm Ready
+                </Button>
+              </>
+            ) : (
+              <Button
+                onClick={() => {
+                  // close modal and reset form values.
+                  close();
+                  reset();
+                  setModalState("init");
+                }}
+                variant="secondary"
+              >
+                Done
+              </Button>
+            )}
+          </div>
+        </ModalWrapper>
+      </Modal>
+    </>
+  );
+};
+
+const SubmitModal = forwardRef(_SubmitModal);
+SubmitModal.displayName = "SubmitModal";
```

Comment:
just a small nit with this file, `SubmitModal` is so generic, took me a while to realize it was doing a specific thing. would be great if we could make this more specific like `SubmitVotesModal` or `CommitVotesModal` or something like that.

Reply:
Good call.